### DISCUSSION TITLE
Abnormal insulation impedance

### DISCUSSION THREAD

**Discussion by Schleppe** [hoylabs/OpenDTU-OnBattery#790] [General]
Hallo,
mein WR HM 1600 4T hatte schon öfters diese Fehlermeldung "Abnormal insulation impedance" und hört auf Strom zu produzieren. Hat jemand eine Ahnung was ich verändern kann, damit dies nicht mehr vorkommt? 
Danke!

**Comment by spcqike** [comment]
> First of all, the abnormal insulation impedance is partly due to the DC cable damage, including the cables between components, the cables between components and the inverter, especially the cables in the corner and the cables laid in the open without pipes, you need to carefully check whether the cables are damaged.

Probleme auf der DC Seite. Von nassen Steckern die anfangen zwischen + und - leitend zu werden, über defekte Kabel (Isolierungen) bis hin zu defekten Paneelen kann es quasi alles sein.

**Reply by Schleppe** [reply]
Danke für deine Antwort - anscheinend gibt es eine Datenbank mit den Fehlercodes. Bitte schick mir den link - Danke!

**Reply by spcqike** [reply]
Meinst du diese?
https://github.com/tbnobody/OpenDTU/blob/master/lib/Hoymiles/src/parser/AlarmLogParser.cpp

**Reply by Schleppe** [reply]
Ja danke - ich dachte, dass vielleicht eine Beschreibung dabei wäre.

**Reply by tonibbb** [reply]
Hallo Schleppe,
bekomme leider auch diese Fehlermeldung (181 - Abnormal insulation impedance). Bist du hier irgendwie weitergekommen? Ich betreibe einen HM-600 erfolgreich an einem 24 V-Akkusystem und will diesen nun durch einen HMS-800 ersetzen und erhalte diese Fehlermeldung sofort nachdem ich die DC-Seite verbinde. Der Wechselrichter beginnt sofort schnell rot zu blinken (wahrscheinlich alle 0,5 s). Laut Bedienungsanleitung entspricht dies "Rotes Blinken (im Abstand von 0,5 s): Fehler der Steuereinheit" oder in einer anderen Bedienungsanleitung: "Rotes Blinken (0,5 s Intervall): AC-Netz ungültig oder Hardwarefehler Weitere Informationen finden Sie auf der Überwachungsplattform von Hoymiles.".  Ich dachte erst es liegt an einem Fehler des Wechselrichters, aber der Fehler tritt auch bei einem 2. HMS-800 auf. DC-seitige Probleme würde ich ausschließen, da der HM-600 funktioniert (außerdem ist alles im Keller in Nähe zum Akku verkabelt, ca. 50 cm). Was da jetzt AC-seitig nicht passen soll (Beschreibung des Fehlers in der Bedienungsanleitung) verstehe ich auch nicht, weil es ja da mit dem HM-600 auch keine Probleme gibt. Hat jemand einen Tipp?

**Reply by tonibbb** [reply]
Oder gibts da noch irgendwas mit diesem neuartigen "Feldstecker"? Habe diesen analog zum Stecker des HM-600 ans Hausnetz angeschlossen. AC-Spannung konnte ich an dem hausseitigen Stecker (bzw. Buchse) messen.

**Reply by SW-Niko** [reply]
Hallo @tonibbb,
Ich würde bei der Fehlermeldung nach einem "Erdschluss-Fehler" suchen. 

In der Anleitung zum HM-300 steht dazu folgendes:
Der Mikro-Wechselrichter verfügt über einen Widerstandssensor zur Messung des Widerstands zwischen
den Ausgängen des PV-Moduls und der Erde. Bei einem Problem mit der Isolierung des PV-Moduls, der
DC-Verkabelung oder den Steckverbindern des Moduls usw. kann der Widerstand zwischen den Ausgängen
des Moduls und der Erde abnehmen. Wenn dieser Widerstand unter einen voreingestellten Schwellenwert
fällt, stellt der Mikro-Wechselrichter die Stromerzeugung ein und meldet diesen Erdschluss. Dieser Fehler
bleibt so lange bestehen, bis er in der S-Miles Cloud (der Überwachungsplattform von Hoymiles) gelöscht
wird. Dieser Fehler bleibt so lange bestehen, bis der Mikro-Wechselrichter neu gestartet wird.
Beachten Sie, dass dieser Fehler nicht gelöscht werden kann, wenn die Fehlerursache weiterhin besteht.
Wenn der Fehler weiterhin besteht, wenden Sie sich an Ihren Installateur oder an Hoymiles, um das Gerät
möglicherweise auszutauschen. 

Warum funktioniert der HM-600 und der HMS-800 nicht? Meine Vermutung ... Das liegt an der internen Messgenauigkeit der Hoymiles.
Die HMS Serie misst etwas genauer oder hat einen strengeren Grenzwert für den Erdschlusswiderstand.

WICHTIG: Der Hoymiles kann nur dann einen Erdschuss vernünftig messen wenn er mit Erde / Schutzleiter verbunden ist.
Das wird über den Schutzleiter am AC-Stecker gemacht + Erdung am Gehäuse. Überprüfe das nochmal (Null-Leiter und Schutz-Leiter vertauscht) oder lass das von einem Fachmann überprüfen. -----230V können tödlich sein! -----

**Reply by Schleppe** [reply]
Ich habe die Kabel vom Akku zum WR getauscht und das Problem war weg. Die Kabel mit der Fehlermeldung waren 1m lang und die neuen Kabel sind 3m lang  - ich weiß nicht ob das einen Zusammenhang hat.

**Reply by SW-Niko** [reply]
Die Länge der Kabel sind in Bezug auf einen Erdschluss irrelevant. Ein Erdschluss entsteht zwischen einem Potential das Erdfrei sein sollte und Erde, meistens wird ein Fehler gemeldet wenn der Erdschluss kleiner als 50KOhm ist. Da spielen ein paar mOhm vom Kabel keine Rolle eher der Isolationswiderstand.

Ich frag jetzt nochmal... Du bist dir sicher das auf der AC Seite alles ok ist? 
Phase, Null-Leiter und Schutz-Leiter, alle drei angeschlossen und nichts vertauscht?

**Reply by tonibbb** [reply]
ich habe drei Kabel, jeweils mit einer Betteri-Buchse (also diese die an der HM-Serie passen) versehen bis zu den Wechselrichtern/Akku liegen, welche jeweils an einem Zählpunkt hängen. An allen dreien konnte ich bisher mit HM-600 bzw. HM-300 problemlos einspeisen. Für den neuen HMS-800 habe ich mir nun ein Adapterkabel Feldstecker auf Betteri gebastelt, aber dieses funktioniert an keinem meiner Einspeisekabel. Klar, müsste alles nochmal überprüfen um Sicherheit zu haben und werde dies auch tun, kann mir aber nicht vorstellen, dass ich versehentlich die Anschlüsse im Adapterkabel vertauscht habe. Ich bin ein bisschen verwirrt ob der unterschiedlichen Fehlerbeschreibungen in den Handbüchern und der Zitierung von spcqike ganz oben, nach der es sich um ein DC-Problem handeln soll. Aber es ist ja interessant wenn Schleppe es durch andere/längere Kabel lösen konnte.

**Reply by tonibbb** [reply]
Eigentlich ist es ja nur wichtig den Schutzleiter richtig anzuschließen. Bei L und N dürfte es nicht entscheident sein.

**Comment by SW-Niko** [comment]
> Eigentlich ist es ja nur wichtig den Schutzleiter richtig anzuschließen. Bei L und N dürfte es nicht entscheident sein.

VORSICHT! Das ist kann egal sein oder auch nicht. Vorsicht ist immer dann geboten wenn am Gerät ein Stecker ist der ein verwechseln mechanisch unterbindet. Hat ein Gerät einen Schukostecker/Eurostecker dann ist das Gerät so aufgebaut, das ein verdrehen von L und N keine negative Auswirkung auf die Funktion und auf die Sicherheit hat.
Hat ein Gerät einen Stecker, der das Vertauschen mechanisch verhindert dann kann ein Verdrehen gefährlich werden.

**Reply by tonibbb** [reply]
Ja, da hast du natürlich recht und mir ist das auch klar. Ich bin in dem Fall (HM bzw. HMS) einfach blind der Annahme gefolgt, weil die Geräte ja mit Schuko-Stecker-Adapertkabel ausgeliefert werden bzw. es welche dafür zu kaufen gibt. Habe jetzt nochmal alle Leiter geprüft und es stimmt alles. L ist L, N ist N und Schutz ist Schutz. Habe nun auch das der Lieferung beiliegende Feldstecker auf Schukostecker Kabel ausprobiert, gleiches Fehlerbild. Bei einem weiteren Startversuch hatte ich nun die DTU ausgeschaltet und nach DC und AC Zuschaltung blinkt es langsam grün! DTU an, keine Fehlermeldung. Da jedoch nichts produziert wurde (Feld gelb und nicht blau) habe ich den WR über die DTU neu gestartet. Ergebnis: WR blinkt wieder rot. Nach weiteren DC Ab- und Zuschaltungen sieht es nun so aus, dass die DTU den rot blinkenden WR zwar erreicht, aber weiterhin auf gelb steht und weder Fehler noch "Wechselrichter gestartet" angezeigt wird. Also weiter suchen und probieren.

**Reply by SW-Niko** [reply]
Gut, Wenn AC seitig alles ok ist dann must du auf der DC Seite nach einer Erdverbindung suchen.
Du hast immer noch die Meldung "Abnormal insulation impedance"? Nicht das wir in die falsche Richtung suchen?

Falls Ja dann versuche mal folgenden einfachen Weg um den Fehler einzugrenzen:
Entferne alle Verbindungen/Komponenten, die nicht zwingend notwendig sind.
Der Inverter sollte auch laufen wenn nur die Batterie angeschlossen ist und die Batterie mit nichts anderem verbunden ist. Auch kein Datenkabel von der Batterie zur DTU. 
Wenn das klappt. dann füge weitere Komponenten und Verbindungen dazu bis der Fehler wieder auftaucht.

**Reply by tonibbb** [reply]
Ich vermute den Erdschluss gefunden zu haben. Das 24 V-Heizelement im Pufferspeicher. Werde das mal wenns dunkel ist und der nicht läuft überprüfen. Interessant nur, dass das den HM-600 nicht stört.

**Reply by rs3hc** [reply]
Klingt für mich so, das wäre der Heizstab defekt oder zumindest fehlerbehaftet,,,

**Reply by SW-Niko** [reply]
Hallo @tonibbb ,
sehr gut. 👍 
Wie gesagt ... es kommt darauf an wie niederohmig dein Erdschluss / Isolations Fehler ist und wie groß die Messtoleranzen der Geräte / Serien sind. Du kannst ja den Heizstab von den 24VDC abstecken und im eingebauten Zustand eine Widerstandsmessung von + gegen Erde und von - gegen Erde machen. Würde mich jetzt interessieren was dabei rauskommt. 🤔 
Ich tippe auf ein paar kOhm.

**Comment by tonibbb** [comment]
Danke für das rege Interesse an meiner Problematik. Hier ein par Messergebnisse: 
Nur der Heizstab, Stromversorgung allpolig entfernt im eingeschraubten Zustand: alle Pole gegen Erde: mehere MOhm, Einschraubung gegen Speicher: overload. Sieht für mich i.O. aus.
Nun zum Akku/Sammelschienen (Heizstab weiterhin getrennt): die etwa 120 12V-Bleiakkus sind je zu zwei Stück in Reihe geschaltet in einem massiven Stahlschrank untergebracht. + und - wird über also über 60 Kabelpaare nach außen geführt und gehen dort auf die Sammelschienen. Die Sammelschienen sind auf Plexiglas befestigt, welches wiederum auf einer Pressspanplatte geschraubt ist, auf welcher auch die Victrons und die HM/HMS befestigt sind. Die Platte ist wiederum von außen an den Stahlschrank geschraubt. Bevor + auf die Sammelschiene geht gehts noch durch jeweils einen LSS durch. Nachdem ich nun alle Kabel die von dieser Anordnung wegführen getrennt habe (Heizstab, AC HM/HMS) steht der Schrank nun quasi auf dem Betonboden im Keller. Die Verbindungen zu den nicht (extra) geerdeten PV-Modulen ist aber noch an die Victrons angeschlossen (draußen ist es dunkel, spielt aber wahrscheinlich auch keine Rolle). Trotzdem messe ich zwischen + und Erde die volle Akkuspannung. Hingegen messe ich wie erwartet nichts wenn ich die Pole eines weiteren nicht zu diesem System gehörenden externen 12 V-Akkus mit Erde verbinde. Nach meinem Verständnis habe ich also irgendwo eine Verbindung zwischen Akku - und Erde. Entweder vielleicht durch irgenwelche Schraubverbindung von der - Sammelschiene an den Schrank (müsste erst recht aufwändig einiges abbauen um das überprüfen zu können) und dann über den Betonboden oder über die Leitungen zu den PV-Modulen bzw. diese selbst? Kann das jemand einschätzen (Plausibilität)?

**Reply by spcqike** [reply]
Ist die - sammelschiene leitend am Schrank angeschraubt? Und ist der Schrank ggf. selber am Boden angeschraubt? 

Ich würde, so wie du es schilderst, auf den Schrank tippen. 

Kannst du ggf DC- der Paneele einfacher abklemmen, als die sammelschiene vom Schrank? Dann könntest du es ja so gegen prüfen, ob es vom Schrank oder von den Modulen her kommen muss.

**Reply by SW-Niko** [reply]
>Trotzdem messe ich zwischen + und Erde die volle Akkuspannung.

Das ist ok.

>Hingegen messe ich wie erwartet nichts wenn ich die Pole eines weiteren nicht zu diesem System gehörenden externen 12 V-Akkus mit Erde verbinde. 

Ich bin mir nicht sicher ob ich das richtig verstanden habe. Du nimmst einen 12V Akku der mit nichts verbunden ist und machst eine Spannungsmessung zwischen dem + Pol und der Erde? In diesem Fall würde ich ich auch mit 0VDC rechnen.

Ich versuch mal zu erklären warum das so ist.
Die einzelne Batterie ist kapazitiv und ohmisch so wenig mit Erde verbunden das du mit dem Messgerät (Mohm-GOhm) den +Pol der Batterie auf das Erdpotential ziehst.
Bei deinem großen Batterieblock gelingt dir das nicht, weil hier die Verbindung Batterie - Erde niederohmiger als das Messgerät ist.

Du kannst folgendes Versuchen. Messe die DC Spannung zwischen dem -Pol und Erde. Ist das unter 1VDC dann kannst du auf Widerstandsmessung umschalten und den Widerstand zwischen -Pol und Erde messen.
Das funktioniert ganz gut weil die Messgeräte zur Widerstandsmessung eine interne 2VDC Quelle benutzen. Das Ergebnis ist nicht zwingend richtig weil sich unter Umständen der Fehlerstrom und der Messstrom vermischen aber du kannst das als Referenz nehmen.
ACHTUNG: Mach das nicht zwischen +Pol und Erde sonst schrottest du dein Messgerät.

**Reply by SW-Niko** [reply]
Noch was .....
Eine Erdverbindung kannst du dir auch über die Steuerung holen. Victron -> DTU -> USB -> PC->Erde

**Reply by tonibbb** [reply]
Spannung zwischen - und Erde sind par mV. Also Widerstandsmessung zwischen - und Erde, Ergebnis: 0,0 Ohm (?!). Ja, Victron -> DTU war angeschlossen, aber 5V-Netzteil für Stromversorgung DTU war getrennt. Schraube als nächstes die ganze Holzplatte samt darauf befindlicher Sammelschienen vom Stahlschrank ab. Nur die Sammelschienen wäre zu aufwendig. Falls das nichts bringt als nächstes dann PV -. Hier noch ein par unerklärliche Messergebnisse: HM und HMS "hängen" in der Luft und AC abgesteckt: + gegen Gehäuse HMS zeigt Akkuspannung. + gegen Gehäuse HM zeigt nur 10 bis 12 V.

**Reply by SW-Niko** [reply]
Oh je .... Das hört sich nach viel Arbeit an.

> HM und HMS "hängen" in der Luft und AC abgesteckt

Du meinst AC abgesteckt und DC angesteckt?  Wenn ja, dann ist die Technologie in den Geräten unterschiedlich. Um einen Erdschluss zu messen benötigt der Hoymiles eine interne Messeinheit gegen Gehäuse/Schutzleiter/Erde. Und das ist vermutlich bei HM und HMS unterschiedlich.

**Reply by tonibbb** [reply]
Problem gelöst. Habe viel auseinandergebaut und dann wieder aufgebaut. Nachdem ich zwischnzeitlich dann auch mal die volle Akkuspannung von + gegen Erde messen konnte, habe ich dann auch noch den Temperaturüberschutzschalter am Hygienespeicher entfernt. Der hat einen Schaden bzw. war zwischen einem der Verbindungen und der metallischen Fühlerhülse niederohmig. Nach dem Wiederzusammenbau habe ich nun keine nennenswerten Spannungen (maximal wenige mV) mehr gegen Erde bzw. gegen den Schrank. Auch wenn ich gerade letzteres nicht wirklich verstanden habe, so bin ich doch nun sehr froh, dass auch der HMS arbeitet. Ich danke für eure Unterstützung und Hilfe.

**Reply by schlimmchen** [reply]
Gratuliere! Danke für die Rückmeldung!

**Reply by SW-Niko** [reply]
Super, Freut mich. 👍 
Immer schön wenn die viele Arbeit belohnt wird.
Du hast mehr oder weniger 0 Ohm zwischen - und Erde gemessen, oder? 
Mich wundert immer noch warum der HM keinen Fehler gemeldet hat.
Ich notiere mal das HM und HMS einen Erdschluss vermutlich mit unterschiedlicher Technologie messen.

**Reply by tonibbb** [reply]
Ja, ich freue mich auch. Ob und wie die messen entzieht sich meiner Kenntnis. Mir ist es ein Rätsel: HM und HMS hängen beide am selben Akku(pack) (24V). Zwischen HM-Gehäuse und + Akku messe ich 24 V und zu - par mV. Zwischen HMS-Gehäuse und - Akku messe ich auch 24 V und zu + 0. Kannst du mir das erklären? Beide Gehäuse sind über den Schutzleiter geerdet (auch wenn die L+N erstmal an eigen Zählern hängt, so sollte ja der Schutzleiter irgendwo gemeinsam in der Erde enden).

**Reply by SW-Niko** [reply]
Richtig, in Privathaushalten ist PE (Schutzleiter) mit N (Nullleiter) verbunden. Es gibt Ausnahmen ... die lassen wir mal außen vor.
![grafik](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129541740/2b420d8c-9d0c-4681-9d04-ccf31c2a6f6a)

Das weitere was jetzt kommt sind Vermutungen von mir.

- Das Hoymiles Gehäuse ist mit PE (Schutzleiter) niederohmig verbunden
- Im Hoymiles ist die AC-Seite von der DC-Seite galvanisch getrennt.
- Die Erdschluß Überwachung benötigt Hoymiles aus technischen Gründen (z.B Personenschutz falls die galvanische Trennung versagt)
- Die Erdschluß Überwachung kann man unterschiedlich realisieren mit Bezug zu + oder Bezug zu -. Deshalb kann man ja nach Ausführung unterschiedliche Spannungen gegen PE messen. Es ja auch egal, weil die Solarpaneel auch floating / potentialfrei sind. 

Und jetzt verwenden wir die Inverter in einem Betrieb, für den sie nicht entwickelt wurden. 

PS: Ich habe beruflich bedingt schon mal eine Erdschlussüberwachung entwickelt darum habe ich etwas Erfahrung.
